rust/lance-linalg/src/distance.rs

```diff
@@ -48,26 +48,37 @@ pub enum DistanceType {
 /// For backwards compatibility.
 pub type MetricType = DistanceType;
 
-pub type DistanceFunc = dyn Fn(&[f32], &[f32]) -> f32 + Send + Sync + 'static;
-pub type BatchDistanceFunc =
-    dyn Fn(&[f32], &[f32], usize) -> Arc<Float32Array> + Send + Sync + 'static;
+pub type DistanceFunc = fn(&[f32], &[f32]) -> f32;
+pub type BatchDistanceFunc = fn(&[f32], &[f32], usize) -> Arc<Float32Array>;
+pub type ArrowBatchDistanceFunc = fn(&[f32], &FixedSizeListArray) -> Arc<Float32Array>;
 
 impl DistanceType {
     /// Compute the distance from one vector to a batch of vectors.
-    pub fn batch_func(&self) -> Arc<BatchDistanceFunc> {
+    pub fn batch_func(&self) -> BatchDistanceFunc {
```

Comment:
Drive by change. It's likely Rust is smart enough to compile away the dynamic dispatch, but just in case I'm replacing with function pointers.

Reply:
Probally worthwhile to separate this change (distance functions) into a separate PR, and run some benchmarks to see any noticeable performance difference?